src/huggingface_hub/hf_api.py

```diff
@@ -1389,6 +1390,13 @@ def list_models(
         cardData: bool = False,
         fetch_config: bool = False,
         token: Optional[Union[bool, str]] = None,
+        library: Optional[Union[str, List[str]]] = None,
+        language: Optional[Union[str, List[str]]] = None,
+        model_name: Optional[str] = None,
+        task: Optional[Union[str, List[str]]] = None,
+        trained_dataset: Optional[Union[str, List[str]]] = None,
+        tags: Optional[Union[str, List[str]]] = None,
+        pipeline_tag: Optional[str] = None,
```

Comment:
I would move those arguments up, between `author` and `search` => this way we start with all possible filters and then have the sort/direction/limit/full/etc. options.
(same below in docstring)